python/python_direct/ops.cpp

```diff
@@ -2350,6 +2349,57 @@ Returns
 -------
 TensorView
     The scattered tensor.
+)",
+      py::return_value_policy::reference);
+  ops.def(
+      "gather",
+      [](TensorView* arg, TensorView* index, int64_t dim) -> TensorView* {
+        NVF_CHECK(
+            arg->nDims() == index->nDims(),
+            "Tensor arguments have different dimensions ",
+            arg->nDims(),
+            " and ",
+            index->nDims());
+        NVF_CHECK(
+            dim >= -arg->nDims() && dim < arg->nDims(),
+            "Tensor arguments have dimension ",
+            arg->nDims(),
+            " so dim argument must satisfy ",
+            -arg->nDims(),
+            " <= dim < ",
+            arg->nDims(),
+            ", but received ",
+            dim);
+        return gather(arg, dim, index);
+      },
+      py::arg("arg"),
+      py::arg("index"),
+      py::arg("dim"),
+      R"(
+Gather values from arg tensor along dim at positions given by index.
+
+The dimension of arg and index must match. For all axes other than dim
+the extent of index in that axis need not be equal to its counterpart
+in arg but must not be greater than it.
+
+Parameters
+----------
+arg : TensorView
```

Comment:
It might be useful to modify this comment.
In `(Ni...,M,Nk...)` it might be helpful to not have M mixed with the Ns.